poly1305/sum_amd64.go

```diff
@@ -20,7 +20,10 @@ func finalize(tag *[TagSize]byte, state *[7]uint64)
 // key allows an attacker to forge messages at will.
 func Sum(out *[16]byte, m []byte, key *[32]byte) {
 	h := newMAC(key)
-	h.Write(m)
+	_, err := h.Write(m)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
The [Hash interface](https://golang.org/pkg/hash/#Hash) explicitly says that `Write` can not return an error:
```go
type Hash interface {
    // Write (via the embedded io.Writer interface) adds more data to the running hash.
    // It never returns an error.
    io.Writer
```
The only reason it returns one is for consistency with io.Writer. So I think you can safely ignore the error here. (Same goes for sum_generic.go and ripemd160.go.)

Reply:
Yes, those were intended as impossible panics; How about a `// Should never panic` or `// Impossible` just to respect independence?

Reply:
I personally think that https://github.com/ProtonMail/crypto/blob/291a78ce599d548a26b6a785a7100bc4757b29a7/poly1305/sum_amd64.go#L14-L17 is a lot of code simply to say "I've looked at this and it can't error". https://github.com/ProtonMail/crypto/blob/af600225e902f2dbacc0b8b028a22431fcdd7201/poly1305/sum_amd64.go#L13-L15 is already fairly obvious in that regard, for anyone that is aware that `Hash.Write` doesn't error. And if they're not, the `Write` method is defined further down in the same file, so I think it's not really necessary to maintain a separation.

Reply:
Removed, this is all upstream code and again out of scope.